cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -25,110 +23,251 @@
+    voting_t voting_;
+};
 
 template <typename Float, typename Task>
 static infer_result<Task> call_daal_kernel(const context_gpu& ctx,
                                            const descriptor_t<Task>& desc,
-                                           const table& data,
+                                           const table& infer,
                                            const model<Task>& m) {
     auto distance_impl = detail::get_distance_impl(desc);
     if (!distance_impl) {
         throw internal_error{ dal::detail::error_messages::unknown_distance_type() };
     }
-    else if (distance_impl->get_daal_distance_type() != detail::v1::daal_distance_t::minkowski ||
-             distance_impl->get_degree() != 2.0) {
+    else if (distance_impl->get_daal_distance_type() != detail::v1::daal_distance_t::minkowski) {
         throw internal_error{ dal::detail::error_messages::distance_is_not_supported_for_gpu() };
     }
 
     auto& queue = ctx.get_queue();
-    interop::execution_context_guard guard(queue);
+    bk::interop::execution_context_guard guard(queue);
 
-    const std::int64_t row_count = data.get_row_count();
-    const std::int64_t neighbor_count = desc.get_neighbor_count();
-
-    auto arr_responses = array<Float>{};
-    auto arr_indices = array<std::int64_t>{};
-    auto arr_distance = array<Float>{};
-
-    const auto daal_data = interop::convert_to_daal_table(queue, data);
+    const auto trained_model = dynamic_cast_to_knn_model<Task, brute_force_model_impl<Task>>(m);
+    const auto train = trained_model->get_data();
+    const auto resps = trained_model->get_responses();
 
-    auto daal_responses = daal::data_management::NumericTablePtr();
-    auto daal_indices = daal::data_management::NumericTablePtr();
-    auto daal_distance = daal::data_management::NumericTablePtr();
-
-    const auto data_use_in_model = daal_knn::doNotUse;
-    daal_knn::Parameter daal_parameter(
-        dal::detail::integral_cast<std::size_t>(desc.get_class_count()),
-        dal::detail::integral_cast<std::size_t>(desc.get_neighbor_count()),
-        data_use_in_model);
+    ONEDAL_ASSERT(train.get_column_count() == infer.get_column_count());
+    const std::int64_t feature_count = train.get_column_count();
+    const std::int64_t train_row_count = train.get_row_count();
+    const std::int64_t infer_row_count = infer.get_row_count();
+    const std::int64_t neighbor_count = desc.get_neighbor_count();
 
+    auto arr_responses = array<std::int32_t>{};
     if (desc.get_result_options().test(result_options::responses)) {
-        arr_responses = array<Float>::empty(queue, 1 * row_count, sycl::usm::alloc::device);
-        daal_responses = interop::convert_to_daal_table(queue, arr_responses, row_count, 1);
+        arr_responses = array<std::int32_t>::empty(queue, infer_row_count);
     }
-    else {
-        daal_parameter.resultsToEvaluate = daal_classifier::none;
+    auto arr_distances = array<Float>{};
```

Comment:
ndarray?

Reply:
I prefer to use array for output cause it is easier to be converted into table